languages/go/pkg/query.go

```diff
@@ -0,0 +1,329 @@
+package oso
```

Comment:
I would probably split this out as a separate host pkg. e.g. github.com/osohq/oso/languages/go/pkg/query. (Potentially github.com/osohq/oso/languages/go/internal/pkg/query)